### ISSUE TITLE
[BUG] env-value-from resolves Secrets/ConfigMaps by name only, ignoring namespace

### ISSUE THREAD

**Issue by darkweaver87** [stackrox/kube-linter#1248]
**System info:**
 - OS: Linux
 - kube-linter 0.8.3

**Describe the bug**
We enabled this check on a whole rendered cluster and got a batch of "unknown key" errors on references that are perfectly valid. It turns out that when two Secrets share a name in different namespaces, one silently replaces the other in the check's lookup, and everything pointing at the loser gets flagged.

**To Reproduce**
1. Save the YAML below — the order matters, `alpha/shared` comes first.
2. Run `kube-linter lint --do-not-auto-add-defaults --include env-value-from collision.yaml`

```
(object: alpha/consumer apps/v1, Kind=Deployment) The container "consumer" is referring
to an unknown key "key-alpha" in secret "shared" (check: env-value-from)
```

Two variants of the same file report nothing, which pins it down: drop `beta/shared`, or just move `alpha/shared` to the end.

**Sample YAML input**
```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: shared
  namespace: alpha
stringData:
  key-alpha: value
---
apiVersion: v1
kind: Secret
metadata:
  name: shared
  namespace: beta
stringData:
  key-beta: value
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: consumer
  namespace: alpha
spec:
  selector:
    matchLabels:
      app: consumer
  template:
    metadata:
      labels:
        app: consumer
    spec:
      containers:
        - name: consumer
          image: registry.example.com/consumer:v1
          env:
            - name: SECRET_VALUE
              valueFrom:
                secretKeyRef:
                  name: shared
                  key: key-alpha
```

**Expected behavior**
`alpha/consumer` reads from `alpha/shared`, which has `key-alpha`, so nothing should be reported — whatever `beta/shared` happens to contain, and in whatever order the documents appear.

**Additional context**
Looks like it comes from `pkg/templates/envvarvaluefrom/template.go` L64-73, where both maps are keyed on the name alone:

```go
secrets[secret.Name] = secret            // Fix: Remove ObjectMeta
configmaps[configmap.Name] = configmap   // Fix: Remove ObjectMeta
```

Keying on `{Namespace, Name}` and matching the referring object's namespace would also close the reverse hole, where a reference to a key that only exists in the shadowing namespace passes today.

Worth mentioning that names like `github` or `credentials` repeat across namespaces all the time, so this shows up quickly once you lint a full cluster in one pass — it cost us 27 false positives across two clusters. #1241 looks like another gap in the same template, happy to help with a PR if the `{Namespace, Name}` direction sounds right.
